com.unity.visualeffectgraph/Runtime/Utilities/Playables/VisualEffectActivationTrackMigration.cs

```diff
@@ -0,0 +1,188 @@
+#if VFX_HAS_TIMELINE && UNITY_EDITOR
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Playables;
+using UnityEngine.Timeline;
+using UnityEngine.VFX;
+
+namespace UnityEditor.VFX.Migration
+{
+    class ActivationToControlTrack : AssetPostprocessor
```

Comment:
Side note: We should investigate for a better solution than AssetPostprocessor, maybe an OnInvalidate on VisualEffectControlTrack  ?

Reply:
Discussiong with @bastienunity about potential better approach (prototyping [here](https://github.com/Unity-Technologies/Graphics/tree/vfx/feature/timeline-scrubbing-without-assetpreprocessor))

Reply:
All in all, about this ⏫, even if the upgrade solution is pretty naive in this PR, it can be reworked afterwards.

Reply:
Well the refactored feature should be shipped along with its upgrade path. So not sure what you mean by reworked afterwards

Reply:
We should be able to do the same without `AssetPostprocessor` (it's never a good thing to rely on this callback for sanitization).

There isn't any sanitize mecanism in Timeline, the usage of AssetPostprocessor is the simple and naïve solution, but not the best (because the callback can't be discarded afterwards, it can't be a long term solution).

A better solution would be to upgrade data directly in `OnValidate` of the track. The problem is, during the first invocation, all clips aren't loaded yet. It can't work out of the box (while PostProcessor is correctly called after loading):
![image](https://user-images.githubusercontent.com/22494706/148950918-858ddb2f-2318-4b6d-a044-362191f491a5.png)
There is potentially a trick forcing a loading (using [ForceReserializeAssets](https://docs.unity3d.com/ScriptReference/AssetDatabase.ForceReserializeAssets.html)) of child clip in track during OnValidate but we aren't completly sure it's legal and it doesn't cause side effect.

Side Node: Initially, we have considered to keep both integration but remove the old one from listing, the problem of this solution comes if we want to do the sanitization later, it's really complex to sanitize director component which are stored in the scene (the bindings are registering a type & guid). Furtheremore, as far as I know, there isn't any way to filter out an existing implemented track from UX.

Reply:
Adding an editor test to check migration at 4b74606e940db9ea34129bfc77579b40c2cb1e2b
![image](https://user-images.githubusercontent.com/22494706/148984370-92138444-466f-4657-9d12-6845f976b029.png)

Reply:
If the issue is the additional callback, we can always use the already existing AssetPostProcessor we already have in the package

Reply:
The problem isn't really the additional callback but the `AssetDatabase.LoadAssetAtPath<TimelineAsset>` which force the loading of every playable after import.
Additionnaly, it won't work if there isn't any import.

We are looking for two alternatives:
1. https://github.com/Unity-Technologies/Graphics/commit/d0da33a029af0f5caecd6da264962c41536e85bc <= Trying to switch the type from Awake in old VisualEffectActivationClip, it can be problematic from AssetDB due to the rentry from an OnValidate. However, it doesn't work so far and it looks hacky, need validation from scriptable & assetDB ⏳
2. https://github.com/Unity-Technologies/Graphics/commit/4d6a3c4d2410c99e8d4d7d964ed292c1edea5ad7 <= Trying to early detect if there is a VisualEffectActivationClip to prevent the loading of TimelineAsset but it doesn't work because `LoadAssetAtPath` doesn't list hidden asset ⏳

Reply:
These prototypes are a deadend, did the minimal smart change at https://github.com/Unity-Technologies/Graphics/commit/11af1e72761af53f27294b3ee93945156445b105
Ideally, we should have a mecanism in Timeline to handle sanitization & upgrade of tracks.